### PR TITLE
Add storage account example and virtual network example

### PR DESCRIPTION
**Pull request by jmspring**
Storage Account example is a command line tool going through the various actions.

The virtual network example was created by Nava Levy, but included here since she didn't have time to generate the pull request.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jmspring__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by ljhljh235** [comment]
@azuresdkci add to whitelist

**Comment by jmspring** [comment]
Note - this patch is part of the CRP Java SDK hackfest that took place the
last three days.

File: azure-mgmt-samples/pom.xml [hunk 1]
```diff
@@ -68,6 +68,11 @@
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>
+            <artifactId>azure-core-test</artifactId>
```
Comment (ljhljh235) [inline azure-mgmt-samples/pom.xml]: Why is this needed?

**Comment by ljhljh235** [comment]
@azuresdkci retest this please

File: azure-mgmt-samples/pom.xml (hunk 1 above)
Comment (jmspring) [inline azure-mgmt-samples/pom.xml]: It was reported by Raj Vengalil during the hackfest that it was needed to
build / work on IntelliJ.

Comment (ljhljh235) [inline azure-mgmt-samples/pom.xml]: It shouldn't. This package only contains one class `MockIntegrationTestBase` which is used by tests to do the recording/mocking. I don't see it being used in any of your changes. 
A cleanup or re-import in IntelliJ might solve the problem. If not, `mvn clean compile` should always tell the truth.

**Comment by jmspring** [comment]
I'll reissue a PR with the broken tests (that this PR adds) fixed.
